python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -279,6 +285,13 @@ def _task_done_callback(task_index, inputs):
             freed = inputs.size_bytes()
             self._metrics.freed += freed
             self._metrics.cur -= freed
+            self._tasks_finished += 1
+            self._tasks_finished_blocks += self._task_blocks_outputted[task_index]
+            self._estimated_output_blocks = round(
+                (self.num_outputs_total() or 1)
```

Comment:
Is `num_outputs_total` equal to the total number of tasks?

Reply:
It's the number of bundles outputted by its input dependency. I guess if blocks are rebundled this can be inaccurate.

Reply:
`num_outputs_total` currently is static, I think we'll also need to update its value to `_estimated_output_blocks`. 
Imagine we have multiple ops (op1 -> op2 -> op3), all of them will output N blocks. And there are M tasks for op1. Then the correct number of final outputs should be M*N^3. If we don't update, we'll get M*N